src/main/transports/electron-offline-net.ts

```diff
@@ -0,0 +1,48 @@
+import { Response, Status, TransportOptions } from '@sentry/types';
+import { logger } from '@sentry/utils';
+import { net } from 'electron';
+
+import { ElectronNetTransport, SentryElectronRequest } from './electron-net';
+import { PersistedRequestQueue } from './queue';
+
+/** Using net module of Electron */
+export class ElectronOfflineNetTransport extends ElectronNetTransport {
+  private _queue: PersistedRequestQueue = new PersistedRequestQueue(30, 30);
+  private _url: string = this._api.getEnvelopeEndpointWithUrlEncodedAuth();
+
+  /** Create a new instance  */
+  public constructor(public options: TransportOptions) {
+    super(options);
+
+    void this.flushQueue();
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public async sendRequest(request: SentryElectronRequest): Promise<Response> {
+    // If this version of Electron doesn't support 'online` detection or it does and it's true
+    if (!('online' in net) || net.online === true) {
+      try {
+        logger.log(`Attempting to send ${request.type}`);
+        const response = await super.sendRequest(request);
+        void this.flushQueue();
+        return response;
+      } catch (e) {
+        logger.warn(e);
+      }
+    }
+
+    await this._queue.add(request);
```

Comment:
We can probably just use a simple binary semaphore to avoid contention here, maybe just have a boolean flag guarding the critical sections (add and pop)?

Reply:
Using a simple condition variable probably works just as well

Reply:
Since `store.update()` is entirely sync, and `queue.add` and `queue.pop` do any update first before performing any async tasks, I don't think there are any data race issues 🤔